app/Http/Controllers/MollieController.php

```diff
@@ -17,43 +18,91 @@
+        $cap = intval($request->cap);
         $total = 0;
+        $requested_method = $request->method;
+        $selected_method = null;
+        $use_fees = config('omnomcom.mollie')['use_fees'];
+        $available_methods = $use_fees ? self::getPaymentMethods() : null;
 
         $orderlines = [];
+        $unpaid_orderlines = OrderLine::query()
+                ->where('user_id', Auth::id())
+                ->whereNull('payed_with_cash')
+                ->whereNull('payed_with_bank_card')
+                ->whereNull('payed_with_mollie')
+                ->whereNull('payed_with_withdrawal')
+                ->orderBy('total_price', 'asc')
+                ->orderBy('created_at', 'desc')
+                ->get();
+
+        if ($unpaid_orderlines->min('total_price') > $cap) {
+            Session::flash(
+                'flash_message',
+                'You cannot complete a purchase using this cap. Please try to increase the maximum amount you wish to pay!'
+            );
+            return Redirect::back();
+        }
 
-        foreach (OrderLine::where('user_id', Auth::id())->whereNull('payed_with_cash')->whereNull('payed_with_bank_card')->whereNull('payed_with_mollie')->whereNull('payed_with_withdrawal')->orderBy('created_at', 'asc')->get() as $orderline) {
-            if ($total + $orderline->total_price > $cap) {
+        foreach ($unpaid_orderlines as $orderline) {
+            if ($total + $orderline->total_price <= $cap) {
+                $orderlines[] = $orderline->id;
+                $total += $orderline->total_price;
+            } else {
                 break;
             }
-            $orderlines[] = $orderline->id;
-            $total += $orderline->total_price;
         }
 
-        if ($total <= 0) {
-            Session::flash('flash_message', 'You cannot complete a purchase using this cap. Please try to increase the maximum amount you wish to pay!');
-            return Redirect::back();
-        }
-
-        $fee = config('omnomcom.mollie')['fixed_fee'] + $total * config('omnomcom.mollie')['variable_fee'];
-
-        $orderline = OrderLine::findOrFail(Product::findOrFail(config('omnomcom.mollie')['fee_id'])->buyForUser(Auth::user(), 1, $fee, null, null, null, 'mollie_transaction_fee'));
-        $orderline->save();
+        if($use_fees){
+            $selected_method = $available_methods->filter(function ($method) use ($requested_method) {
+                return $method->id === $requested_method;
+            });
 
-        $orderlines[] = $orderline->id;
+            if ($selected_method->count() === 0) {
+                return Redirect::back()->with('flash_message','The selected payment method is unavailable, please select a different method');
+            }
 
-        $transaction = self::createPaymentForOrderlines($orderlines);
+            $selected_method = $selected_method->first();
+        
+            if (
+                $total < floatval($selected_method->minimumAmount->value) ||
```

Comment:
Float conversion here does not seem like a good idea. This could lead to rounding errors causing ambiguous behavior. If possible, avoid it. Although for this maybe the OrderLine models should be migrated to use integer types instead of floats. For being a rarely used feature, it could be okay.